rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -546,7 +563,12 @@ class Client : public ClientBase
         "Received feedback for unknown goal. Ignoring...");
       return;
     }
-    typename GoalHandle::SharedPtr goal_handle = goal_handles_[goal_id];
+    typename GoalHandle::SharedPtr goal_handle = goal_handles_[goal_id].lock();
+    // Forget about the goal if there are no more user references
+    if (!goal_handle) {
+      goal_handles_.erase(goal_id);
```

Comment:
This should print something - error / info / debug. Something to indicate that you wrongfully got feedback

Reply:
I don't think this is a "wrong" case. It's possible for a user to "fire-and-forget" a goal and I think we would be in this case. I.e. The user send a goal, but doesn't bother retaining the goal handle (maybe they don't care about a result or feedback). Logging a debug message whenever we remove a goal handle isn't a bad idea though.

Reply:
That would also work.

Reply:
Logging debug in d06e8e0